source/include/localisation/cs_TrackableParser.h

```diff
@@ -8,45 +8,126 @@
 
 #include <ble/cs_iBeacon.h>
 #include <events/cs_EventListener.h>
-#include <set>
-#include <util/cs_Coroutine.h>
+#include <protocol/cs_TrackableParserPackets.h>
+#include <localisation/cs_TrackingFilter.h>
+
 
 /**
  * Transforms EVT_DEVICE_SCANNED and EVT_ADV_BACKGROUND_PARSED
  * into EVT_TRACKING_UPDATE events.
  *
- * Responsible for throttling input to the localisation module, which
- * may take up more computation resources.
- *
- * Suggestion: This component seems like a natural place to handle
- * identification when ids rotate by generating a (crowntone) mesh
- * local identifier as an abstraction (and size reduction) layer.
+ * Responsible for throttling input to the localisation module using filter parsers.
  */
 class TrackableParser : public EventListener {
 public:
 	void init();
 	void handleEvent(event_t& evt);
 
+	constexpr static size_t MAX_FILTER_IDS = 8;
+	constexpr static size_t FILTER_BUFFER_SIZE = 512;
+
 private:
+	// -------------------------------------------------------------
+	// ------------------ Advertisment processing ------------------
+	// -------------------------------------------------------------
+
+	void handleScannedDevice(scanned_device_t* device);
+
 	/**
 	 * Dispatches a TrackedEvent for the given advertisement.
 	 */
 	void handleBackgroundParsed(adv_background_parsed_t *trackableAdvertisement);
 
+	// -------------------------------------------------------------
+	// --------------------- Filter buffer data --------------------
+	// -------------------------------------------------------------
+
+	/**
+	 * The raw byte buffer in which TrackingFilters are allocated.
+	 * Memory is managed as a stack:
+	 * - the top is administrated by _filterBufferEndIndex
+	 * - new filters are allocated at the top (if space allows)
+	 * - when a filter is removed, all filters on top of it are relocated
+	 *   immediately to avoid gaps.
+	 */
```

Comment:
Good idea. It's basically a preallocated object pool, and considering our memory management guidelines it makes sense to factor this out into a class.I think it would make for a great isolated task or do you want to take that on for this pull request?

Reply:
A separate task / PR would be fine yeah.

Reply:
Done: https://www.pivotaltracker.com/story/show/177380657